### PR TITLE
[Feature] Add ResignLeadership to plan execution

### PR DESCRIPTION
**Pull request by ajanikow**

### CODE REVIEW

File: pkg/deployment/reconcile/action_resign_leadership.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
+//
+// DISCLAIMER
+//
+// Copyright 2020 ArangoDB GmbH, Cologne, Germany
```
Comment (ewoutp) [inline pkg/deployment/reconcile/action_resign_leadership.go:4]: 2021?

File: pkg/deployment/reconcile/action_resign_leadership.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Ewout Prangsma
```
Comment (ewoutp) [inline pkg/deployment/reconcile/action_resign_leadership.go:20]: Really?

File: pkg/deployment/reconcile/action_resign_leadership.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	a.actionImpl = newActionImplDefRef(log, action, actionCtx, shutdownMemberTimeout)
+
+	return a
+}
+
+type actionResignLeadership struct {
+	actionImpl
+}
+
+func (a *actionResignLeadership) Start(ctx context.Context) (bool, error) {
```
Comment (ewoutp) [inline pkg/deployment/reconcile/action_resign_leadership.go:53]: Missing a comment here

File: pkg/deployment/reconcile/action_resign_leadership.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		return false, false, errors.WithStack(err)
+	}
+
+	c, err := a.actionCtx.GetDatabaseClient(ctx)
+	if err != nil {
+		log.Debug().Err(err).Msg("Failed to create member client")
+		return false, false, errors.WithStack(err)
+	}
+
+	jobStatus, err := arangod.CleanoutServerJobStatus(ctx, m.CleanoutJobID, c, agency)
```
Comment (ewoutp) [inline pkg/deployment/reconcile/action_resign_leadership.go:122]: Is this correct?
We're not doing cleanout

Comment (ajanikow) [reply pkg/deployment/reconcile/action_resign_leadership.go:127]: It is valid option - Cleanout and Reasign jobs are handled in same way, just trigerred in different way.

Function could be called just FetchJobStatus.

**Review by joerg84 — APPROVED** [review]

**Pull request merged**